apps/server/src/routes/session.ts

```diff
@@ -23,4 +48,71 @@ export const SessionRoute = new Elysia({ prefix: "/session" })
       return session;
     },
     { params: t.Object({ sessionId: t.String() }) },
+  )
+  .post(
+    "/",
+    async ({ error, body, user, workspace }) => {
+      const station = await getStation(db, body.stationId);
+      if (station === undefined) {
+        return error(404, "Station not found");
+      }
+      const unit = await getUnitBySerialNumber(db, body.serialNumber);
+      if (unit === undefined) {
+        return error(404, `Serial Number ${body.serialNumber} not found`);
+      }
+      const toInsertSession = {
+        unitId: unit.id,
+        userId: user.id,
+        projectId: station.projectId,
+        stationId: body.stationId,
+        integrity: body.integrity,
+        aborted: body.aborted,
+        notes: body.notes,
+        commitHash: body.commitHash,
+      };
+      const newSession = await createSession(db, toInsertSession);
+      if (newSession.isErr()) {
+        return error(500, newSession.error);
+      }
+
+      // Create all the measurements for this session
+      await Promise.all(
+        body.measurements.map(async (measurement) => {
+          // NOTE: TestId should be provided in the DS
+          let test = await getTestByName(db, measurement.name);
+          if (test === undefined) {
+            const testResult = await createTest(db, {
+              name: measurement.name,
+              projectId: station.projectId,
+              measurementType: measurement.data.type,
+            });
+            if (testResult.isErr()) {
+              return error(500, testResult.error);
+            }
```

Comment:
This doesn't actually return an error from the route because its inside of the `map` callback, the proper way to do this would be to use a transaction with a regular for loop, and throw an error to abort the transaction, but catch it with `fromPromise` to handle as a `Result`.
Something like
```typescript
const res = fromPromise(ctx.db.transaction((tx) => {
  for (const meas of body.measurements) {
    ...
    if (testResult.isErr()) {
      throw testResult.error;
    }
  }
}));
if (res.isErr()) {
  return error(500, res.error);
}
```